catkin_ws/src/vision/src/object_detection.py

```diff
@@ -166,7 +165,7 @@ def detect_on_image(raw_img, camera_id):
         obj.extra_field = obj.extra_field if obj.extra_field is not None else -1234.5
     
 
-    detectionFrameArray = cleanDetections(detectionFrameArray, confidence)
+    detectionFrameArray = cleanDetections(detectionFrameArray)
```

Comment:
since you're changing the arguments were passing to cleanDetections you'll also have to modify that function